python/src/aac/persistence/load.py

```diff
@@ -0,0 +1,57 @@
+"""Module for loading data from the state file and making it available to the rest of the AaC package."""
+
+import json
+
+from aac import __version__
+from aac.io.paths import sanitize_filesystem_path
+from aac.lang.language_context import LanguageContext
+from aac.persistence.state_file_error import StateFileError
+
+
+def get_language_context_from_state_file(file_name: str) -> LanguageContext:
+    """
+    Return the language context loaded from the named state file.
+
+    Args:
+        file_name (str): The name of the state file.
+
+    Returns:
+        The language context after loading definitions from the files stored the state file.
+    """
+    language_context = LanguageContext()
+    version, files, definitions, plugins = decode_state_file(file_name)
+
+    if version != __version__:
+        raise StateFileError(
+            f"Version mismatch: State file written using version {version}; current AaC version {__version__}"
+        )
```

Comment:
❤️ This is a great idea! I wonder if we could expand on this to provide "support" version traversal (e.g. 0.1.4 -> 0.1.5 is allowed, but 0.1.4 -> 0.2.0 isn't.) That seems like a lot of overhead for current user levels; maybe in the future.